### ISSUE TITLE
[CRITICAL] "Hello World" Tutorial is Broken

### ISSUE THREAD

**Issue by lazaridiscom** [EOSIO/eos#5126]
(This issue is CRITICAL because of the very negative user-experience)

## Problem

Multiple Users have reported since months problems with the "Hello World" tutorial:

https://developers.eos.io/eosio-cpp/docs/hello-world

The main problems are:

* eosiocpp is not found
  * SOLVED: use `sudo ./eosio_install.sh`, which creates symlinks, thus the binaries are found without further PATH variables.
* library files are not found (The introduced regression was fixed, see #5133)

## Reproduction

* https://developers.eos.io/eosio-nodeos/docs/autobuild-script
* https://developers.eos.io/eosio-nodeos/docs/install-executables
* https://developers.eos.io/eosio-cpp/docs/hello-world

### Cmd Sequence

```bash
git clone https://github.com/EOSIO/eos.git --recursive
cd eos
./eosio_build.sh
sudo ./eosio_install.sh
eosiocpp -n hello
cd hello
eosiocpp -o hello.wast hello.cpp
```

### Workaround

not needed anymore

```bash
# get eosiocpp and other binaries into the path, for current cmd-window
export PATH=/usr/local/eosio/bin:$PATH
```

```bash
# permanently add path
echo 'PATH=$PATH:/usr/local/eosio/bin/' >> ~/.profile
# reload profile to activate path for current session
source ~/.profile
```

## Related Issues / Resources

https://eosio.stackexchange.com/questions/1719/can-not-find-eosiolib-eosio-hpp-when-compiling-with-eosiocpp-2

#5040
#5044 
#5129

**Comment by ChristianDunst** [comment]
Are you using a docker image to develop?

**Comment by lazaridiscom** [comment]
See "Reproduction" section, this affects mostly users which go the non-docker way. Although latest problems (e.g. #5133) occur with docker, too. Workaround is now available, team should provide the final solutions, thus future users have less problems.

**Comment by MrToph** [comment]
Would love to see an updated tutorial using [eosio.wasmsdk](https://github.com/EOSIO/eosio.wasmsdk).
 Especially as the way described in the official documentation has been deprecated as of 1.2.0 and will be removed in 1.3.0

**Comment by lazaridiscom** [comment]
@MrToph , you can track this within #5040.

**Comment by lazaridiscom** [comment]
Path and library issues were fixed by the team (version 1.2.0), closing.
